Content.Shared/Contraband/ContrabandSystem.cs

```diff
@@ -48,6 +58,11 @@ private void OnExamined(Entity<ContrabandComponent> ent, ref ExaminedEvent args)
         if (!_contrabandExamineEnabled)
             return;
 
+        if(_contrabandExamineOnlyInHudEnabled)
+            if (TryComp<InventoryComponent>(args.Examiner, out var comp))
+                if (_inventorySystem.TryGetContainerSlotEnumerator(new Entity<InventoryComponent?>(args.Examiner, comp), out var inventorySlot, SlotFlags.EYES))
+                    if(inventorySlot.NextItem(out var item))
+                        if (!TryComp<ShowJobIconsComponent>(item, out var jComp)) return;
```

Comment:
ok a few things:
- combine these if clauses using `&&`
- there are relay events for inventory items so you don't have to loop over all your inventory slots like this
- should probably be its own component instead of the `ShowJobIconsComponent`
- Conceptually I'm not sure if this should be hidden from players. I assume it was intended to be shown for everyone to give players clarity about the contraband status of items. Without it it will cause arguments between security and other crew about what is and is not contraband, especially since the contraband list was removed from the WizDen server rules. However, with a cvar forks can enable this if desired, so it might be ok.

Reply:
> Conceptually I'm not sure if this should be hidden from players. I assume it was intended to be shown for everyone to give players clarity about the contraband status of items.

I 100% agree with this. I thought the original intent for the contraband status to give regular players knowledge about what they have. Security already has a list of contraband, they don't need to see the contraband status of items.

Reply:
> * there are relay events for inventory items so you don't have to loop over all your inventory slots (and it seems you dont even loop over the whole inventory, but only check the first item)

can you please tell me how to do this?  I tried looking through the available methods but didn't find the one I needed, so I did it with nextItem in the hope that the character wouldn't have more slots with the EYES flag


> Conceptually I'm not sure if this should be hidden from players. I assume it was intended to be shown for everyone to give players clarity about the contraband status of items. Without it it will cause arguments between security and other crew about what is and is not contraband, especially since the contraband list was removed from the WizDen server rules. However, with a cvar forks can enable this if desired, so it might be ok.

this is exactly why I created a corresponding cvar, maybe it's better to make it false by default, so as not to make changes to the configuration of the wizden servers

Reply:
> can you please tell me how to do this?

For an example, look at how `SlipAttemptEvent` checks if the player wears galoshes or magboots to decide if the slip should succeed or not.
- You define a new `GetContrabandDetailEvent` and raise it on the player examining the item
- you define an inventory relay in `InventorySystem.Relay.cs`
- you define a new `ShowContrabandDetailsComponent` and add it to the HUD in yaml
- you make a new system for the component and subscripe to the relay event, setting a varable to true so that the OnExamined function knows that it should print the text
- ahosts should also have the component so that they can always read the text without wearing a HUD